### PR TITLE
Add Logging and detector mask

### LINKED ISSUE #2: mask and calibrant do nothing
Add the ability to view the mask and calibrant on the plot

### PR DESCRIPTION
**Pull request by tizayi**
- Adds some logging
- Adds a detector mask
- Corrects error in calculation of s
- Small changes to the layout

resolves #2

### CODE REVIEW

File: src/plot/centrePlot.tsx [hunk 1]
```diff
@@ -460,18 +480,16 @@ function getRange(): (state: ResultStore) => UnitRange | null {
...
-          break;
-        default:
-          result = unit(value, state.qUnits);
+
+      if(state.requested === ScatteringOptions.d){
+        return convertBetweenQAndD(unit(value, state.dUnits));
+      }
+
+      if(state.requested === ScatteringOptions.s){
+        return convertFromSTooQ(unit(value, state.sUnits));
```
Comment (stan-dot) [inline src/plot/centrePlot.tsx:489]: nice

File: src/plot/legendBar.tsx [hunk 2]
```diff
@@ -27,6 +27,7 @@ export default function LegendBar(): JSX.Element {
           <Typography variant="h6"> Legend</Typography>
           <Divider />
           <FormGroup>
+            {/* Detector */}
```
Comment (stan-dot) [inline src/plot/legendBar.tsx:30]: nice, I wish more codebases did this

File: src/plot/plotStore.ts [hunk 3]
```diff
@@ -29,6 +31,8 @@ export interface PlotConfig {
 export const usePlotStore = create<PlotConfig>((set) => ({
   detector: true,
   detectorColor: { r: 144, g: 19, b: 254, a: 0.4 },
```
Comment (stan-dot) [inline src/plot/plotStore.ts:33]: could: extract rgba color into a typed value and name the value by the color name

File: src/plot/svgMask.tsx [hunk 4]
```diff
@@ -0,0 +1,139 @@
+import { SvgRect } from '@h5web/lib';
+import type { SVGProps } from 'react';
+import { Vector3 } from 'three';
+
+type Rect = [Vector3, Vector3];
+
+interface Props extends SVGProps<SVGPathElement> {
```
Comment (stan-dot) [inline src/plot/svgMask.tsx:7]: could: I am slightly opinionated to the convention where this interface is called `SvgMaskProps`, in a case it needs to be used elsewhere later on.

File: src/plot/svgMask.tsx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    coords: Rect;
+    strokePosition?: 'inside' | 'outside';
```
Comment (stan-dot) [inline src/plot/svgMask.tsx:9]: could: consider making it a boolean? or enum

File: src/plot/svgMask.tsx [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+function SvgMask(props: Props){
+    const [detectorUpper, detectorLower] = props.coords;
+    // check if clone is needed later on
+    const fulllength = detectorLower.clone().sub(detectorUpper);
+    const gaplength = fulllength.clone().multiply(props.gapFraction);
+    const moduleLength = 
+        fulllength.clone()
+        .sub(gaplength.clone()
+        .multiply(props.numModules.clone().subScalar(1)))
+        .divide(props.numModules);
```
Comment (stan-dot) [inline src/plot/svgMask.tsx:25]: should: add more explaining comments - when I look at it the first time I am not sure what is the system in question these equations model

File: src/presets/presetManager.ts [hunk 7]
```diff
@@ -11,13 +11,21 @@ import {
...
  */
 export function createInternalDetector(detectorData: IODetector): AppDetector {
   return {
+    mask: {
+      horizontalModules: 1,
+      verticalModules: 1,
+      horizontalGap: 0,
+      verticalGap: 0,
+      missingModules: []
+    },
```
Comment (stan-dot) [inline src/presets/presetManager.ts:28]: should: extract this into internalDetectorMask

File: src/results/rangeTable.tsx [hunk 8]
```diff
@@ -15,7 +15,7 @@ import { ScatteringOptions, useResultStore } from "./resultsStore";
 import { ReciprocalWavelengthUnits, WavelengthUnits } from "../utils/units";
 import {
   convertBetweenQAndD,
```
Comment (stan-dot) [inline src/results/rangeTable.tsx:17]: maybe: could this 'between' be also renamed to 'to', like on line 18? it's clearer

File: src/results/rangeTable.tsx [hunk 9]
```diff
@@ -118,11 +119,11 @@ export default function RangeTable(props: { qRange: UnitRange }): JSX.Element {
...
                 >
-                  <MenuItem value={WavelengthUnits.nanometres}>
-                    {WavelengthUnits.nanometres}
+                  <MenuItem value={ReciprocalWavelengthUnits.nanometres}>
+                    {"1 / nm"}
                   </MenuItem>
-                  <MenuItem value={WavelengthUnits.angstroms}>
-                    {"\u212B"}
+                  <MenuItem value={ReciprocalWavelengthUnits.angstroms}>
+                  {"1 / " + "\u212B"}
```
Comment (stan-dot) [inline src/results/rangeTable.tsx:126]: must: extract the angstorm string into a constant

File: src/results/resultsBar.tsx [hunk 10]
```diff
@@ -25,54 +25,63 @@ import RangeTable from "./rangeTable";
...
-      diagramFull = fullQrange.apply(convertBetweenQAndS).to("nm");
-      diagramRequested = UnitRange.fromNumericRange(
+      ).to("nm")
+    };
+  }
+
+  if(resultStore.requested === ScatteringOptions.s){
+    return {
+      textBoxUnits :resultStore.sUnits,
+      diagramVisible:visableQRange.apply(convertFromQTooS).to("nm^-1"),
```
Comment (stan-dot) [inline src/results/resultsBar.tsx:68]: should: fix inconsistent linting

File: src/results/resultsBar.tsx [hunk 11]
```diff
@@ -25,54 +25,63 @@ import RangeTable from "./rangeTable";
...
-      textBoxUnits = resultStore.qUnits;
+      ).to("nm^-1")
+  }
   }
-  return { textBoxUnits, diagramVisible, diagramFull, diagramRequested };
+
+  return { 
+    textBoxUnits:resultStore.qUnits,
+    diagramVisible:visableQRange.to("nm^-1"),
+    diagramFull: fullQrange.to("nm^-1") ,
```
Comment (stan-dot) [inline src/results/resultsBar.tsx:80]: should: extract the unit into a string, 

and then pack in inside an enum (unless the library does not allow, I don't quite remember)

**Review by stan-dot — CHANGES_REQUESTED** [review]
small code clarity and maintainability changes

File: src/results/resultsBar.tsx (hunk 11 above)
Comment (tizayi) [reply src/results/resultsBar.tsx:80]: This is a good point as the enum already exists.

File: src/plot/svgMask.tsx (hunk 5 above)
Comment (tizayi) [reply src/plot/svgMask.tsx:9]: This was to match SvgRect in h5web/lib.

https://github.com/silx-kit/h5web/blob/main/packages/lib/src/interactions/svg/SvgRect.tsx

File: src/dialogs/detectorDialog.tsx [hunk 12]
```diff
@@ -60,8 +59,7 @@ export default function DetectorDialog(props: {
             <Grid item xs={12} sm={12} md={12} lg={7}>
               <DetectorTable />
             </Grid>
-            <Grid item md={0} lg={1} >
-            </Grid>
+            <Grid item md={0} lg={1}></Grid>
```
Comment (stan-dot) [inline src/dialogs/detectorDialog.tsx:62]: why leave an empty grid element here? whitespace can be achieved with `margin` css properties

File: src/plot/plotStore.ts [hunk 13]
```diff
@@ -7,6 +7,15 @@ export enum PlotAxes {
...
 
+// default colours used in app
+const lilac = { r: 144, g: 19, b: 254, a: 0.4 };
+const black = { r: 0, g: 0, b: 0, a: 1 };
+const blackOpaque = { r: 0, g: 0, b: 0, a: 0.2 };
+const turquoise = { r: 80, g: 227, b: 194, a: 0.4 };
+const mustard = { r: 245, g: 166, b: 35, a: 1 };
+const green = { r: 65, g: 117, b: 5, a: 1 };
+const red = { r: 208, g: 2, b: 27, a: 1 };
+
```
Comment (stan-dot) [inline src/plot/plotStore.ts:18]: good extraction for now

the detectorColor property of PlotConfig should be considered to change into string
react-color is not well-maintained - last update 4 years ago
https://www.npmjs.com/package/react-color

using raw hex values with an extension like this (https://marketplace.visualstudio.com/items?itemName=kamikillerto.vscode-colorize) should also provide also excellent developer experience

regarding the input itself it is not necessary to use a react library, a raw html element can be considered

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/color

still this is beyond the scope of this PR, I'd suggest making a new issue from this comment and decide what to do later

File: src/plot/svgMask.tsx [hunk 14]
```diff
@@ -1,139 +1,154 @@
...
+  // Get full length and width of detector
+  const fulllength = detectorLower.clone().sub(detectorUpper);
+  // Get the length of the horizontal and vertical gaps between modules
+  const gaplength = fulllength.clone().multiply(props.gapFraction);
+  // Get the length and width of each module
+  // module length =
+  //    (detector length - (number of gaps)*(gap length))/(number of modules)
+  const moduleLength = fulllength
+    .clone()
+    .sub(gaplength.clone().multiply(props.numModules.clone().subScalar(1)))
```
Comment (stan-dot) [inline src/plot/svgMask.tsx:29]: can we get some comment what does this do?

File: src/presets/presetManager.ts [hunk 15]
```diff
@@ -9,8 +9,17 @@ import {
   IOBeamline,
   IODetector,
   IOPresetConfig,
+  DetectorMask,
 } from "../utils/types";
+import { LengthUnits, WavelengthUnits } from "../utils/units";
 
+const DefaultDtectorMask: DetectorMask = {
```
Comment (stan-dot) [inline src/presets/presetManager.ts:16]: should: fix variable name typo, and either all caps or lower, not PascalCase

**Review by stan-dot — CHANGES_REQUESTED** [review]
besides some typo, all is good

**Review by stan-dot — APPROVED** [review]
lgtm

**Pull request merged**